util/http/http_handler.h

```diff
@@ -80,12 +81,22 @@ class HttpListenerBase : public ListenerInterface {
     enable_metrics_ = true;
   }
 
-  // Enforce authentication to this HTTP server with the user "user"
-  // and this password.
+  // Enforce authentication to this HTTP server with the user username_
+  // and pass token password_.
+  void SetUsername(std::string username) {
+    username_ = std::move(username);
+  }
+
+  // Enforce authentication to this HTTP server with the user username_
+  // and pass token password_
   void SetPassword(std::string password) {
     password_ = std::move(password);
   }
 
+  void SetSkipAuthForPath(std::function<bool(std::string_view path)> func) {
```

Comment:
First, I didn't know you could specify parameter names in function signatures. Nice!
Second, I'd try to make this a unified API. Maybe 2 callbacks: `ShouldAuth(path)` and `TryAuth(user, pass)`
WDYT?
I guess that could be in another PR, but it'd be easier to pull helio once.

Reply:
@royjacobson suggested to have a single API: `Auth(path, user, pass)`
It's a much better idea imho :)

Reply:
We don't really need `TryAuth(user, pass)` since I added `SetUsername` on the listener. That means, that both the `password` and `username` can be set by the consumers of this lib (in our case dragonfly). `SkipAuthForPath` allows to skip authorization on certain paths, e.g, `/metrics` (which is what we need) and `SetUsername` will allow us to auth against a pair of username/password set by the users of `HttpListenerBase`. 

Maybe I am missing something in your suggestion so I am all eyes :eyes: 

p.s. I had to google what WDYT is :rofl: It means what do you think right? :tong

Reply:
With a callback you can allow multiple users to authenticate to the same HTTP endpoint. I thought it's what we wanted for the ACL stuff, no?

Reply:
@royjacobson Good question and I am sorry I was not clear about this earlier on the daily. I had a discussion with @romange yesterday and we agreed that currently, admin should only be used by the `default` user. There is no current plan to integrate ACL with admin port or HTTP atm

Reply:
It's still a better API for future expansion, no?

Reply:
I mean yes but I can't really foresee if this is something that we need and I would rather wait than introducing dead code (in a sense that is not used/executed anywhere)

Reply:
I thought `Auth(path, user, pass)` is what we discussed - being able to whitelist a specific path and to support multiple "user/pass" pairs (where one of them is default/pass). If it somehow was not clear, I am sorry. 
so +1 to a `(path, user, pass)` functor

Reply:
done